macros/src/lib.rs

```diff
@@ -34,17 +34,22 @@ impl DerivedTS {
             .export
             .then(|| self.generate_export_test(&rust_ty, &generics));
 
-        let export_to = {
+        let output_path_fn = {
             let path = match self.export_to.as_deref() {
                 Some(dirname) if dirname.ends_with('/') => {
                     format!("{}{}.ts", dirname, self.ts_name)
                 }
                 Some(filename) => filename.to_owned(),
-                None => format!("bindings/{}.ts", self.ts_name),
+                None => format!("{}.ts", self.ts_name),
             };
 
             quote! {
-                const EXPORT_TO: Option<&'static str> = Some(#path);
+                fn output_path() -> Option<std::path::PathBuf> {
+                    let path = std::env::var("TS_RS_EXPORT_DIR");
+                    let path = path.as_deref().unwrap_or("./bindings");
+
+                    Some(std::path::Path::new(path).join(#path))
+                }
```

Comment:
The implementation of that is super simple - Take `TS_RS_EXPORT_DIR` (or `./bindings`), and append the name (or `#[ts(export_to = "...")]` override to it.